src/reify.ml4

```diff
@@ -11,9 +11,29 @@ let _ = Goptions.declare_bool_option {
   Goptions.optread = (fun () -> !cast_prop);
   Goptions.optwrite = (fun a -> cast_prop:=a);
 }
-
+                                     
 let is_cast_prop () = !cast_prop                     
-                     
+
+let share_strings = ref (false)
+let _ = Goptions.declare_bool_option {
+  Goptions.optsync = true; Goptions.optdepr = false;
+  Goptions.optname = "Maximal sharing of ascii/strings";
+  Goptions.optkey = ["Template";"Share";"Strings"];
+  Goptions.optread = (fun () -> !share_strings);
+  Goptions.optwrite = (fun a -> share_strings:=a);
+}
```

Comment:
Indeed it should be a net improvement always. What will quote (quote a) do? Let-binding all the string literals is doable but a bit more complicated I think, I'll leave that as another improvement. I'll first finish this by making all Quote Definitions not go through pretyping.